.travis.yml

```diff
@@ -42,38 +20,57 @@ before_install:
   - |
     if [[ "$TRAVIS_OS_NAME" == "linux" ]]
     then
-        ./scripts/verify-commit-messages.sh "$TRAVIS_COMMIT_RANGE" \
-        && ./scripts/bootstrap-ubuntu-14-04.sh
+        ./scripts/verify-commit-messages.sh "$TRAVIS_COMMIT_RANGE" || exit 1;
+        ./scripts/bootstrap-ubuntu-14-04.sh;
     fi
   - |
     if [[ "$TRAVIS_OS_NAME" == "osx" ]]
     then
-        ./scripts/bootstrap-osx.sh
-    fi
-
-install:
-  - |
-    if [[ "$TRAVIS_RUST_VERSION" == nightly ]]
-    then
-        rustup component add clippy-preview
+        ./scripts/bootstrap-osx.sh;
     fi
 
 script:
   - cargo build --verbose
   - cargo test --verbose
-  - |
-    if [[ "$TRAVIS_RUST_VERSION" == nightly ]]
-    then
-        cargo clippy --verbose --all --tests
-    fi
   - cargo doc
 
-after_success:
-  # measure code coverage and upload to coveralls.io
-  - |
-    if [[ "$TRAVIS_OS_NAME" == "linux" ]] && [[ "$TRAVIS_RUST_VERSION" == stable ]]
-    then
-      cargo install cargo-travis || echo "cargo-travis has been already installed"
-      export PATH=$HOME/.cargo/bin:$PATH
-      cargo coveralls
-    fi
+matrix:
+  allow_failures:
+    - rust: beta
+    - rust: nightly
+  include:
+    - os: linux
+      rust: nightly
+      env:
+        NAME: clippy
+      install:
+        - rustup component add clippy-preview
+      script:
+        - cargo clippy --verbose --all --tests
+
+    - os: linux
+      rust: stable
+      sudo: required
+      env:
+        NAME: coverage
+      addons: # needed for `cargo install cargo-travis`
+        apt:
+          packages:
+            - libcurl4-openssl-dev
+            - libelf-dev
+            - libdw-dev
+            - binutils-dev
+            - cmake
+          sources:
+            - kalakris-cmake
+      cache:
+        directories:
+          - /home/travis/.cargo
+          - /home/travis/build/tox-rs/tox/target/kcov-master
```

Comment:
`cache cargo` does almost the same: https://docs.travis-ci.com/user/caching/#rust-cargo-cache
Why cache only `kcov-master` inside `target`?

Reply:
https://levans.fr/rust_travis_cache.html

Reply:
> Why cache only kcov-master inside target?

Because we don't need to cache target/debug because it will be rebuilt for sure. So it's a waste of time and the resources of travis